src/util/address.rs

```diff
@@ -212,6 +212,11 @@ impl Address {
         Address(self.0 - size)
     }
 
+    /// Apply an signed offset to the address.
+    pub const fn offset(self, offset: isize) -> Address {
+        Address((self.0 as isize).wrapping_add(offset) as usize)
```

Comment:
We may use [wrapping_add_signed](https://doc.rust-lang.org/std/primitive.usize.html#method.wrapping_add_signed) introduced since Rust 1.66.0

```suggestion
        Address(self.0.wrapping_add_signed(offset))
```

Actually, when we compute an address at an offset from a given one, we are not expecting overflow.  It's better if it has undefined behavior in the release build and panics in the debug build.  Unfortunately, the standard library doesn't have such a method.  If we are pedantic, we can use [overflowing_add_signed](https://doc.rust-lang.org/std/primitive.usize.html#method.overflowing_add_signed) (since 1.66.0) instead and assert the boolean return value is false.

```suggestion
        let (result, overflow) = self.0.overflowing_add_signed(offset);
        debug_assert!(!overflow, "Overflowed when adding offset {offset} to {self}");
        Address(result)
```

Reply:
Thanks. I will use `wrapping_add_signed`.